docs/src/reviewing_guide.md

```diff
@@ -0,0 +1,95 @@
+
+Another, less obvious, source of untrusted data are third party crates. Changes to dependencies should be monitored closely.
+
+So what are some things to look out for when dealing with untrusted data?
+1. Any parsing of untrusted data can panic, return an error or generate the wrong data. If the code parsing is inside the
+Tari codebase, it should have matching tests and be fuzzed. If the code is in a third party crate, it should be reviewed very
+carefully. Does the third party crate have fuzzing? Does it have tests? Is it actively maintained? 
+    1. Be careful when using crates that wrap C libraries and other native code. A panic in native code can not be recovered 
+and could be used to crash a node.
+2. Be especially careful when reading the length for a buffer or Vec from untrusted data. If the length is too large, it could
+cause an out of memory exception. If a length is received from an untrusted source, it should be checked against a maximum before allocating memory.
+   1. E.g. `let length = read_u64(stream); let mut buf = vec![0u8; length];` should be replaced with `let mut buf = vec![0u8; min(length, MAX_LENGTH)];`
+
+
+### Comparing to Diagrams
+When reviewing a pull request, it is important to understand the context of the change. Arguably the easiest way to
+achieve this is to review it in the context of the [diagrams](todo). 
+
+Some questions to ask:
+
+1. Firstly, is the diagram up to date with the code?
+2. Does the PR require the diagram to be updated?
+
+### Catching Common Mistakes
+1. Using `unwrap` or `expect` in production code. These should only be used in tests and in code that is not reachable in production.
+2. As stated before, watch out for parsing of untrusted data, especially when using a third party crate.
+3. When parsing untrusted data it is extremely important that the code is fuzzed and that all branches of the code are covered by tests. Coverage reports are generated whenever code is commited to the development branch, but
+PRs will need to be checked manually if they change this code.
+4. What happens when functions are called with no data, too much data, repeated data, etc.?
+5. Can some flow of data lead to a panic?
+6. When using mutexes, readwrite locks or semaphores, can the lock be held for a long time?
+7. When requesting data from another node, there are a few things to look out for:
+   1. What happens if the node does not respond? Is there a timeout? The timeout should be set as low as possible, otherwise 
+the node can hold up the processing on the requesting node.
+   2. What happens if the node returns unexpected data?
+8. In cases where we have received data from a node, whether we requested it or not, if it is bad, is the node banned?
+
+
+### More specific things to look out for
+
+#### Usize
+`usize` is a platform dependent type. It is 32 bits on 32 bit platforms and 64 bits on 64 bit platforms. This can cause 
+differences in hashing and serialisation. It should be avoided.
```

Comment:
```suggestion
differences in hashing and serialisation. It should be avoided when sending data between agents.
```
Working with usize internally is preferred as its more performant, and as such its becomes very much a pain to use a iterator/counter as anything else as usize.

Reply:
We could add `It should be avoided when sending data between agents or when the code can be compiled for both 32 and 64-bit platforms.`

Reply:
I don't know if it is more performant, do you have a link to substantiate?

Reply:
I don't think adding "when sending data" is correct, since there are cases where you might calculate a field incorrectly locally